src/main/java/dao/H2Factory/H2UserDao.java

```diff
@@ -0,0 +1,116 @@
+package dao.H2Factory;
+
+import customerproductorder.models.User;
+import dao.DaoException;
+import dao.UserDaoInterface;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+
+class H2UserDao implements UserDaoInterface {
+
+    private final ConnectionProvider connectionProvider;
+
+    protected H2UserDao(ConnectionProvider connectionProvider) {
+        this.connectionProvider = connectionProvider;
+    }
+
+    public void create(User newUser) throws DaoException {
+        try (Connection connection = connectionProvider.getConnection()) {
+            PreparedStatement st = connection.prepareStatement("insert into "
+                    + "customerapplication.account(username,password) "
+                    + "values(?,?)");
+            st.setString(1, newUser.getUsername());
+            st.setString(2, newUser.getPassword());
+            st.executeUpdate();
+            connectionProvider.destroy();
+        } catch (SQLException ex) {
+            throw new DaoException("Inserting data error", ex);
+        }
+    }
+
+    public User get(int id) throws DaoException {
+        User gotUser = null;
+        try (Connection connection = connectionProvider.getConnection()) {
+            PreparedStatement st = connection.prepareStatement("select * from "
+                    + "customerapplication.account where id=?");
+            st.setInt(1, id);
+            ResultSet rs = st.executeQuery();
+            connectionProvider.destroy();
```

Comment:
I would recommend to destroy connection manager after you finished to work with ResultSet. Actually you can have some time scrollable resultsets - that can require connection to DB.